tests/compute_orchestration/test_crud_operations.py

```diff
@@ -0,0 +1,161 @@
+from clarifai.client.nodepool import Nodepool
+from clarifai.client.user import User
+
+NOW = uuid.uuid4().hex[:10]
+
+CREATE_COMPUTE_CLUSTER_USER_ID = os.environ["CLARIFAI_USER_ID"]
+CREATE_COMPUTE_CLUSTER_ID = f"ci_test_compute_cluster_{NOW}"
+CREATE_NODEPOOL_ID = f"ci_test_nodepool_{NOW}"
+CREATE_DEPLOYMENT_ID = f"ci_test_deployment_{NOW}"
+
+COMPUTE_CLUSTER_CONFIG_FILE = "tests/compute_orchestration/configs/example_compute_cluster_config.yaml"
+NODEPOOL_CONFIG_FILE = "tests/compute_orchestration/configs/example_nodepool_config.yaml"
+DEPLOYMENT_CONFIG_FILE = "tests/compute_orchestration/configs/example_deployment_config.yaml"
+
+CLARIFAI_PAT = os.environ["CLARIFAI_PAT"]
+
+
+@pytest.fixture
+def client():
+  return User(user_id=CREATE_COMPUTE_CLUSTER_USER_ID, pat=CLARIFAI_PAT)
+
+
+@pytest.fixture
+def create_compute_cluster():
+  return ComputeCluster(
+      user_id=CREATE_COMPUTE_CLUSTER_USER_ID,
+      compute_cluster_id=CREATE_COMPUTE_CLUSTER_ID,
+      pat=CLARIFAI_PAT)
+
+
+@pytest.fixture
+def create_nodepool():
+  return Nodepool(
+      user_id=CREATE_COMPUTE_CLUSTER_USER_ID, nodepool_id=CREATE_NODEPOOL_ID, pat=CLARIFAI_PAT)
+
+
+@pytest.mark.requires_secrets
+class TestComputeOrchestration:
+  """Tests for the Compute Orchestration resources.
+    CRUD operations are tested for each of the following resources:
+    - compute cluster
+    - nodepool
+    - deployment
+    """
+
+  @classmethod
+  def setup_class(cls):
+    """Setup: Clean up any pre-existing resources before tests."""
+    cls.client = User(user_id=CREATE_COMPUTE_CLUSTER_USER_ID, pat=CLARIFAI_PAT)
+    cls.compute_cluster = create_compute_cluster
+    cls.nodepool = create_nodepool
+    cls._cleanup_resources()
+
+  @classmethod
+  def teardown_class(cls):
+    """Teardown: Clean up any resources created during the tests."""
+    cls._cleanup_resources()
+
+  @classmethod
+  def _cleanup_resources(cls):
+    """Helper function to delete any existing resources."""
+    try:
+      cls.nodepool.delete_deployments([CREATE_DEPLOYMENT_ID])
+    except Exception:
+      pass  # Ignore if not found
+
+    try:
+      cls.compute_cluster.delete_nodepools([CREATE_NODEPOOL_ID])
+    except Exception:
+      pass  # Ignore if not found
+
+    try:
+      cls.client.delete_compute_clusters([CREATE_COMPUTE_CLUSTER_ID])
+    except Exception:
+      pass  # Ignore if not found
+
+  def test_create_compute_cluster(self, client, caplog):
+    with open(COMPUTE_CLUSTER_CONFIG_FILE) as f:
+      config = yaml.safe_load(f)
+    config["compute_cluster"]["id"] = CREATE_COMPUTE_CLUSTER_ID
+    with open(COMPUTE_CLUSTER_CONFIG_FILE, "w") as f:
+      yaml.dump(config, f)
+    with caplog.at_level(logging.INFO):
+      client.create_compute_cluster(
+          compute_cluster_id=CREATE_COMPUTE_CLUSTER_ID,
+          config_filepath=COMPUTE_CLUSTER_CONFIG_FILE)
+      assert "Compute Cluster created" in caplog.text
+
+  def test_create_nodepool(self, create_compute_cluster, caplog):
```

Comment:
I'm just checking if we used a minimal node pool configuration and followed [this approach](https://clarifai.atlassian.net/wiki/spaces/EN/pages/3693150243/Compute+Orchestration+Uploading+your+first+model+to+dev#:~:text=NOTE%3A%20set%20min%20instances%20to%200%20so%20that%20when%20you%E2%80%99re%20not%20testing%20things%20will%20scale%20down%20to%20zero%20nodes%20and%20not%20cost%20money.%20You%20can%20experiment%20with%20min%20instances%20%3E%200%20to%20make%20sure%20nodes%20stay%20up%20in%20the%20dev%20compute%20plane%20and%20speed%20up%20cold%20starts%2C%20but%20that%20will%20cost%20money%20until%20you%20delete%20the%20nodepool.) for testing node pools. Since these tests will be frequent, a minimal setup should be sufficient for testing CRUD operations.

Reply:
Yes, I have kept all the instances count max to 1 in nodepool and deployments